### ISSUE TITLE
[java] UnnecessaryVarargsArrayCreation: False positive when removing array creates overload ambiguity

### ISSUE THREAD

**Issue by yegor256** [pmd/pmd#6611]
## Affects PMD Version

7.23.0

## Rule

`category/java/bestpractices.xml/UnnecessaryVarargsArrayCreation`

## Description

`UnnecessaryVarargsArrayCreation` flags `new String[]{...}` passed to a varargs parameter as unnecessary, but in some cases the explicit array is **required** to disambiguate between overloaded methods/constructors. Removing it produces a compile error.

## Minimal reproducer

```java
public final class Demo {
    public Demo(final String file, final String... lines) {}
    public Demo(final String... args) {}

    public static void main(final String[] a) {
        new Demo("fibonacci", new String[]{"a", "b"}); // PMD flags this
    }
}
```

PMD reports:

```
Unnecessary explicit array creation for varargs method call (UnnecessaryVarargsArrayCreation)
```

But applying the suggested fix (removing `new String[]`) does not compile:

```java
new Demo("fibonacci", "a", "b");
```

```
Demo.java:6: error: reference to Demo is ambiguous
        new Demo("fibonacci", "a", "b");
        ^
  both constructor Demo(String,String...) in Demo
  and constructor Demo(String...) in Demo match
```

## Expected behavior

The rule should not report a violation when the target type has another overload whose parameter types would also be applicable after the array-creation is expanded into varargs, because in that case the explicit array is load-bearing for overload resolution.

## Actual behavior

The rule reports the array creation as unnecessary without checking for overload ambiguity, leading users to a non-compiling fix.

## Context

Reported downstream in qulice: https://github.com/yegor256/qulice/issues/1525
